chore: update the roadmap (#584)
CeresDB/ceresdb: docs/guides/src/dev/roadmap.md

- [x] Static cluster mode with `CeresMeta`.
- [x] Basic implementation of hybrid storage format.

### v0.4.0
- [x] Implement more sophisticated cluster solution that enhances reliability and scalability of CeresDB.
- [x] Set up nightly benchmark with TSBS.

### v1.0.0-alpha (Released)
- [x] Implement Distributed WAL based on `Apache Kafka`.
- [x] Release Golang client.
- [x] Improve the query performance for traditional time series workloads.
- [x] Support dynamic migration of tables in cluster mode.

### v1.0.0
- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.

> - [ ] Finish the majority of work related to PartitionTable.

Reviewer: ```suggestion
- [ ] Finish the majority of work related to Table Partitioning.
```

> - [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. Theses works includes:

Reviewer: ```suggestion
- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. These works includes:
```

>     - Multi-tier cacheing.

Reviewer 2: ```suggestion
    - Multi-tier cache.
```
Or fix the typo:
```suggestion
    - Multi-tier caching.
```

    - Introduce various methods to reduce the data fetched from remote storage (improve the accuracy of SST data filtering).
    - Increase the parallelism while fetching data from remote object-store.

>     - Add some resource-controlling methods.

Reviewer 2: It seems this does nothing with the query performance improvement.

- [x] Static cluster mode with `CeresMeta`.
- [x] Basic implementation of hybrid storage format.

### v0.4.0
- [x] Implement more sophisticated cluster solution that enhances reliability and scalability of CeresDB.
- [x] Set up nightly benchmark with TSBS.

### v1.0.0-alpha (Released)
- [x] Implement Distributed WAL based on `Apache Kafka`.
- [x] Release Golang client.
- [x] Improve the query performance for traditional time series workloads.
- [x] Support dynamic migration of tables in cluster mode.

### v1.0.0

> - [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.

Reviewer 2: ```suggestion
- [ ] Formally release CeresDB and its SDKs with all breaking changes finished.
```

- [ ] Finish the majority of work related to `Table Partitioning`.
- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. These works includes:
    - Multi-tier cache.
    - Introduce various methods to reduce the data fetched from remote storage (improve the accuracy of SST data filtering).
    - Increase the parallelism while fetching data from remote object-store.

> - [ ] Improve data ingestion performance with some resource-controlling methods.

Reviewer 2: ```suggestion
- [ ] Improve data ingestion performance by introducing resource control over compaction.
```

### Afterwards
With an in-depth understanding of the time-series database and its various use cases, the majority of our work will focus on performance, reliability, scalability, ease of use, and collaborations with open-source communities.

> - [ ] Add utilities that support promQL, influxQL, opentsdb protocol, and so on.

Reviewer: ```suggestion
- [ ] Add utilities that support `PromQL`, `InfluxQL`, `OpenTSDB` protocol, and so on.
```

> - [ ] Provide basic utilities for operation and maintenance. Specifically, include the following:

Reviewer 2: ```suggestion
- [ ] Provide basic utilities for operation and maintenance. Specifically, the following are included:
```

>     - Deployment tools that fit well for both Kubernetes and bare servers.

Reviewer: ```suggestion
    - Deployment tools that fit well for both `Kubernetes` and `Bare` servers.
```

Author: The word bare is not terminology.
